Sources/CasePaths/EnumReflection.swift

```diff
@@ -131,19 +136,43 @@ private func isUninhabitedEnum(_ type: Any.Type) -> Bool {
 }
 
 private func enumTag<Case>(_ `case`: Case) -> UInt32? {
-  let metadataPtr = unsafeBitCast(type(of: `case`), to: UnsafeRawPointer.self)
+  guard let (metadataPtr, vwt) = enumBits(for: Case.self) else { return nil }
+  return withUnsafePointer(to: `case`) { vwt.getEnumTag($0, metadataPtr) }
+}
+
+private func associatedValues<Enum, Values>(of `enum`: Enum) -> Values {
+  let (metadataPtr, vwt) = enumBits(for: Enum.self)!
+  let enumPtr = UnsafeMutablePointer<Enum>.allocate(capacity: 1)
+  enumPtr.initialize(to: `enum`)
+  let untypedPtr = UnsafeMutableRawPointer(enumPtr)
+  vwt.destructiveProjectEnumData(untypedPtr, metadataPtr)
+  // Maybe this should be .bindMemory instead of .assumingMemoryBound?
+  let valuesPtr = untypedPtr.assumingMemoryBound(to: Values.self)
```

Comment:
It think what you have with `assumingMemoryBound(to:)` is the right call. I do wonder if it's possible to do fewer raw-typed pointer dances by describing the value witness table slightly differently, but I'm not sure if that is safe to do.

```suggestion
  let valuesPtr = untypedPtr.assumingMemoryBound(to: Values.self)
```